fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java

```diff
@@ -2110,6 +2175,9 @@ private LoanScheduleDTO rescheduleNextInstallments(final MathContext mc, final L
         // this block is to retain the schedule installments prior to the
         // provided date and creates late and early payment details for further
         // calculations
+        if (Objects.isNull(rescheduleFrom) && loanApplicationTerms.isDownPaymentEnabled()) {
```

Comment:
Why do we need this change?

Reply:
if the rescheduling happens because down payment is enabled, but interest recalculation is disabled, in this case the rescheduleFrom date will be `null` and the rescheduling will fail with exception